### PR TITLE
Match response keys with Android

### PR DESCRIPTION
**Pull request by bbirman**
First step to match changes here https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2095

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSURLResponse+SFAdditions.m [hunk 1]
```diff
@@ -37,7 +37,8 @@ - (NSDictionary *)asDictionary {
     responseDictionary[@"textEncodingName"] = self.textEncodingName;
     responseDictionary[@"URL"] = self.URL.absoluteString;
     if ([self isKindOfClass:[NSHTTPURLResponse class]]) {
-        responseDictionary[@"allHeaderFields"] = ((NSHTTPURLResponse *)self).allHeaderFields;
+        responseDictionary[@"headers"] = ((NSHTTPURLResponse *)self).allHeaderFields;
```
Comment (bbirman) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSURLResponse+SFAdditions.m:40]: I noticed that the header name capitalization is different, Android is "content-type" vs iOS "Content-Type", should we make the iOS header keys lowercase?

Comment (bhariharan) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSURLResponse+SFAdditions.m:40]: Nah - should be fine.

Comment (wmathurin) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSURLResponse+SFAdditions.m:40]: It's also Content-Type on [Android](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestResponse.java#L52), as it [should be](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type), isn't it?

Comment (bbirman) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSURLResponse+SFAdditions.m:40]: Interesting -- when going through .getAllHeaders they're lowercase
image

Comment (wmathurin) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSURLResponse+SFAdditions.m:40]: They are supposed to be case insensitive according to https://stackoverflow.com/a/5259004/11461823
Looking at the implementation on okHttp, it's indeed ignoring the case when getting headers by name: https://github.com/square/okhttp/blob/master/okhttp/src/main/kotlin/okhttp3/Headers.kt#L102

**Pull request merged**